docs/architecture-decisions/ADR_001_project_generator_architecture.md

```diff
@@ -0,0 +1,275 @@
+
+#### 3. Project Management
+- **Feature Status Tracking**: While in an AlgoKit project, users SHALL be able to determine status of what features have been added to the project
+- **Version Tracking**: The project generator SHALL be able to know what version of the project generator was used to generate the project
+- **Compatibility Management**: The project generator SHALL be able to know if there are compatibility issues between the project generator used to generate the project and the project generator being used to add a feature. The project generator SHALL flag the user to these compatibility issues or error if the compatibility issues are not reconcilable
+
+#### 4. Non Functional Requirements
+- **No Manual Dependency Installation**: The project generator SHALL NOT require a user to manually install a python dependency or any other required dependencies in order to run the project generator
+
+### Current Gaps
+
+**Diff Preview and Approval System**
+The current implementation applies changes directly without showing users what will be modified. There is no mechanism to preview changes before they are applied, and no approval workflow exists. This makes users hesitant to run generators on existing projects as they cannot predict the impact.
+
+**Intelligent Conflict Detection**
+While Copier provides basic file overwrite warnings, there is no intelligent detection of feature conflicts. The system cannot determine if a feature has already been added by examining project configuration or file patterns. This can lead to duplicate configurations or broken setups when features are added multiple times.
+
+**Comprehensive Compatibility Management**
+The current system has basic version requirement checking but lacks a compatibility matrix between generator versions. There are no automated migration paths when upgrading generators, and no clear way to handle breaking changes between versions. Projects can become stuck on old generator versions with no upgrade path.
+
+**Self-Contained Execution Environment**
+Users must manually install Python and other dependencies before using AlgoKit. The typed client generators require separate package installations (`@algorandfoundation/algokit-client-generator` for TypeScript, `algokit-client-generator` for Python). This creates barriers to entry and potential version conflicts.
+
+**Scattered Feature Status Tracking**
+Feature configuration is distributed across multiple files (`.copier-answers.yml`, `.algokit.toml`, `pyproject.toml`, `package.json`). There is no single command to view all enabled features and their status. Users must manually inspect multiple files to understand their project configuration.
```

Comment:
This is very typical in all projects, so it's pretty normal to be navigating to a relevant file depending on what context specific thing you are doing.

Quite often I only need to know configuration information about a specific facet of my project, so most of the time I'm ok with looking in the relevant file.

Do you have an activity in mind where you'd want info collated from various configurations? Would it include all the info or only specific fields? What fields are important?